packages/nns/src/governance.canister.ts

```diff
@@ -335,18 +334,16 @@ export class GovernanceCanister {
       principal,
     );
 
-    const from_subaccount = fromSubAccount && Uint8Array.from(fromSubAccount);
-
     // Send amount to the ledger.
-    await icrcLedgerCanister.transfer({
```

Comment:
Don't we need to check wether `this.hardwareWallet`?

Then revert to using `transfer`?

Even though it's supported in the latest version, I don't think that if we use the icrc1 transfer, it will show as "Stake Neuron" in the device.

Would you plan to check for that in the nns-dapp and tell the HW user to upgrade?

Reply:
So as it turns out, the Ledger app doesn't support making ICRC-1 transfers to the governance canister. At all, with or without stake neuron flag.
But I think the NNS-dapp is the right place to make the decision which function to call.
We're just going to have to have 2 separate stake function for a lot longer than I was hoping for.
Does that sound good?

Note that this PR doesn't affect that. This function was already doing an ICRC-1 transfer and it still is, just through a different code path.

Reply:
I know what happened. It works for principals, but not for subaccounts because we gave them the old textual encoding.

Let me create a new ticket for them to support the new ticket and ther Stake Neuron for ICRC1.

Reply:
I created two tickets for Zondax. The problem was that when we implemented the ICRC1, we had a different textual representation. Sending to principals works, because they are the same textual representation in the old standard. But not to subaccounts.

* https://github.com/Zondax/ledger-icp/issues/216
* https://github.com/Zondax/ledger-icp/issues/217

Reply:
Thanks for filing those issues. But they are different issues than what I described.
You literally can't send to a principal that isn't 29 bytes long:
https://github.com/Zondax/ledger-icp/blob/9f964dc4c014dfb5a9be6f6f0cc281a60003e088/app/src/candid/candid_parser.c#L350
Just try sending to `ttwhn-ittl6-fnuml-tqdgz-tbyhu-f7jti-ft2cr-znzsu-biv4l-gpvg4-ba` and you'll see what I mean.

And do you want to wait for those issues to be fixed before we merge this PR?
Once again, note that this PR is unrelated. This PR only changes the code path through which we create the same transaction.
But if you want any changes before merging, or you don't think the check for whether to use this function based on HW should be in the (NNS-dapp) application rather than the library, let me know.